### ISSUE TITLE
CsvFileReader#readAllAsSequence fails on non-equal sized rows

### ISSUE THREAD

**Issue by koral--** [doyaaaaaken/kotlin-csv#61]
**Describe the bug**
If rows are not equal-sized an exception is thrown:
```com.github.doyaaaaaken.kotlincsv.util.CSVFieldNumDifferentException: Fields num seems to be 4 on each row, but on 2th csv row, fields num is 3.```

**To Reproduce**
```kotlin
        csvWriter().open(csvFile) {
            writeRow(listOf("a"))
            writeRow(listOf("a", "b"))
        }
        csvReader().open(csvFile) {
           readAllAsSequence()
        }
```

**Expected behavior**
Missing cells are treated as nulls or empty strings.

**Environment**
 - kotlin-csv version 0.11.0
 - OS: Android 10

**Screenshots**
N/A

**Comment by doyaaaaaken** [comment]
@koral-- 
Thanks for reporting!

Throwing exception is expected behavior.
If you change behavior, the `skipEmptyLine` option on CsvReader  is available.
See https://github.com/doyaaaaaken/kotlin-csv#customize

**Comment by doyaaaaaken** [comment]
Any other question on this topic? Feel free to ask.
If not, I'll close this issue.

**Comment by koral--** [comment]
I don't longer need this library, but can help resolve this issue if needed.

I don't understand your explanation. Could you elaborate?
Is it really intended that CSV written by this library is not readable by itself?

**Comment by doyaaaaaken** [comment]
The "CSV" definition is defined on [RFC4180](https://tools.ietf.org/html/rfc4180).
It says here.
> Within the header and each record, there may be one or more fields, separated by commas.  Each line should contain the same number of fields throughout the file.

So, [your sample code](https://github.com/doyaaaaaken/kotlin-csv/issues/61#issue-692202304) doesn't write "CSV" file, so it's not unnatural that kotlin-csv cannot read your writing file.

**Comment by koral--** [comment]
OK, thanks. Now I understand the explanation.
One little question then: is it intended that library allows writing a file which does not conform to CSV format?
In this particular case the 2nd `#writeRow()` call may throw an exception (or it can be thrown when closing file, technical details are not important now).

**Comment by doyaaaaaken** [comment]
> One little question then: is it intended that library allows writing a file which does not conform to CSV format?

Yes, it's intended.

> In this particular case the 2nd #writeRow() call may throw an exception (or it can be thrown when closing file, technical details are not important now).

Yes, it should be. To throw exception on the 2nd `writeRow()` call is useful.
At this point, I don't plan to implement this validation by myself, but pull requests are welcomed. 👍

**Comment by koral--** [comment]
OK so I guess that this issue can be rephrased to something like "Fail fast on writing malformed CSV" and tagged as enhancement., right?

**Comment by doyaaaaaken** [comment]
Yeah, right. Thank you.
Maybe, it'd be better to create another issue, relate this issue on that, and close this issue.
